repose-aggregator/components/services/container-configuration-service/impl/src/main/scala/org/openrepose/nodeservice/containerconfiguration/ContainerConfigurationServiceImpl.scala

```diff
@@ -63,9 +63,27 @@ class ContainerConfigurationServiceImpl @Inject()(@Value(ReposeSpringProperties.
     configurationService.unsubscribeFrom(ContainerConfigurationFilename, this)
   }
 
-  override def getVia: Optional[String] = {
+  override def getRequestVia: Optional[String] = {
     initializationCheck()
-    Optional.ofNullable(patchedDeploymentConfiguration.getVia)
+    Option(patchedDeploymentConfiguration.getViaHeader)
+      .map(viaHdr => Optional.ofNullable(viaHdr.getRequestPrefix))
```

Comment:
I like the third one best and it makes adding the interim deprecated one easy too.